primerj/src/main/java/org/primer/primerj/core/TxBuilder.java

```diff
@@ -614,19 +606,14 @@ public int compare(Out out1, Out out2) {
             value += out.getOutValue();
         }
 
-        boolean needAtLeastReferenceFee = TxBuilder.needMinFee(tx.getOuts());
-
         List<Out> bestCoinSelection = null;
         Out bestChangeOutput = null;
         while (true) {
             long fees = 0;
 
-            if (lastCalculatedSize >= 1000) {
-                // If the size is exactly 1000 bytes then we'll over-pay, but this should be rare.
-                fees += (lastCalculatedSize / 1000 + 1) * Utils.getFeeBase();
+            if (lastCalculatedSize >= 0) {
+                fees += lastCalculatedSize * (Utils.getFeeBase() / 1000);
```

Comment:
best use assignment here instead of increment, note right above it's initialized to 0.